sdk/core/src/main/java/org/apache/mesos/config/ConfigurationComparator.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.mesos.config;
+
+/**
+ * This interface defines a way to compare two {@link Configuration} objects for equality. This
+ * interface is intentionally separate from {@link Object#equals(Object)} in order to have very
+ * explicit requirements around equality checks when they're needed.
+ *
+ * @param <T> The {@code Configuration} type that will be parsed.
+ */
+public interface ConfigurationComparator<T extends Configuration> {
```

Comment:
Comparator is slightly different, it also requires `int compare(T a, T b)`

This is why I'd gone with 'Comparer' to begin with but that creates its own confusion.